core/DataAccess/LogAggregator.php

```diff
@@ -432,11 +433,15 @@ protected function getSelectDimensions($dimensions, $tableName, $appendSelectAs
 
             if (!is_numeric($selectAs)) {
                 $selectAsString = $selectAs;
-            } else {
-                // if function, do not alias or prefix
-                if ($this->isFieldFunctionOrComplexExpression($field)) {
-                    $selectAsString = $appendSelectAs = false;
+            } else if ($this->isFieldFunctionOrComplexExpression($field)) {
```

Comment:
can you add a test for this?

Reply:
Will add a unit/integration test. the system tests should show a difference (I haven't copied the files over)

Reply:
Added a test

Reply:
just reading https://dev.mysql.com/doc/refman/8.0/en/select.html ... the `AS` is optional. I suppose we don't bother about this for now until this is a use case? Be best I think... 

does the regex need to check it case insensitive?

also just double checked there cannot be two AS statements in there cause we have an entry for each dimension... just fyi ... was wondering about that.

Reply:
This PR isn't about requiring the `AS`, it's about detecting when it's there and just using it instead of assigning a new one. It's a workaround for when a complex sql expression is used as a dimension, which currently doesn't work since it just gets duplicated in both the select & group by parts.

Since it has to be used explicitly by a plugin developer, I think we don't have to check for uniqueness of the AS, since there will be an error and it will be programmer error.

Should be a case insensitive check, will look into that.

Reply:
> This PR isn't about requiring the AS, 

I meant if someone used `foo bar` we wouldn't detect the `bar` alias. All good though and not needed to be detected right now.